### DISCUSSION TITLE
Using the /rss.xml/route.ts syntax returns 404

### DISCUSSION THREAD

**Discussion by askides** [vercel/next.js#51948] [App Router]
Hello, I'm building an RSS Feed and a Sitemap.xml file for my app which is multi-tenant, but using the dot syntax with the extension (following the doc here: https://nextjs.org/docs/app/building-your-application/routing/router-handlers#non-ui-responses) the app return 404.

Instead, without extension everything works.

What it could be?

Version of Next.js: "13.4.7", using app router.

Thanks!

**Comment by ozguruysal** [comment]
You should add the extension to folder name.

**Comment by bolechen** [comment]
Sorry, I know this post might be old, but I'm also facing this issue and haven't been able to solve it after a whole day. Did you eventually find the answer?

**Reply by bolechen** [reply]
I was incredibly stupid. The issue wasn't with Next.js App Router handling .xml routes or middleware configuration. The real
  culprit was a single line in .vercelignore: **/*.xml which excluded ALL XML files from deployment, including our route folders
  like src/app/api/feed2.xml/ and src/app/rss.xml/.

  Vercel's build logs showed the routes were never built because the files were ignored during deployment. After commenting out the  problematic line, the XML routes should deploy properly.
